internal/services/recommendation_poller.go

```diff
@@ -151,29 +151,34 @@ func PollForRecommendations(msg *kafka.Message, consumer_object *kafka.Consumer)
 		log.Errorf("Error while checking for recommendation_set record: %s", err)
 		return
 	}
-	recommendationFound := !reflect.ValueOf(recommendation_stored_in_db).IsZero()
+	workloadExists := model.WorkloadExistsByID(workloadID)
 
-	switch recommendationFound {
-	case false:
-		poll_cycle_complete := requestAndSaveRecommendation(kafkaMsg, "New")
-		if poll_cycle_complete {
-			commitKafkaMsg(msg, consumer_object)
-		}
-		// To consume upcoming Kafka msg, explicitly
-		return
-	case true:
-		// MonitoringEndTime.UTC() defaults to 0001-01-01 00:00:00 +0000 UTC if not set
-		if !recommendation_stored_in_db.MonitoringEndTime.UTC().IsZero() {
-			duration := maxEndTimeFromReport.Sub(recommendation_stored_in_db.MonitoringEndTime.UTC())
-			if int(duration.Hours()) >= cfg.RecommendationPollIntervalHours {
-				poll_cycle_complete := requestAndSaveRecommendation(kafkaMsg, "Update")
-				if poll_cycle_complete {
+	if workloadExists { // Housekeeper may wipe workload record by the time poller requests for a recommendation
+		recommendationFound := !reflect.ValueOf(recommendation_stored_in_db).IsZero()
+
+		switch recommendationFound {
+		case false:
+			poll_cycle_complete := requestAndSaveRecommendation(kafkaMsg, "New")
+			if poll_cycle_complete {
+				commitKafkaMsg(msg, consumer_object)
+			}
+			// To consume upcoming Kafka msg, explicitly
+			return
+		case true:
+			// MonitoringEndTime.UTC() defaults to 0001-01-01 00:00:00 +0000 UTC if not set
+			if !recommendation_stored_in_db.MonitoringEndTime.UTC().IsZero() {
+				duration := maxEndTimeFromReport.Sub(recommendation_stored_in_db.MonitoringEndTime.UTC())
+				if int(duration.Hours()) >= cfg.RecommendationPollIntervalHours {
+					poll_cycle_complete := requestAndSaveRecommendation(kafkaMsg, "Update")
+					if poll_cycle_complete {
+						commitKafkaMsg(msg, consumer_object)
+					}
+				} else {
 					commitKafkaMsg(msg, consumer_object)
 				}
-			} else {
-				commitKafkaMsg(msg, consumer_object)
 			}
+			return
 		}
-		return
 	}
```

Comment:
Shouldn't we have else block here and commit the kafka message? because autocommit is set to false in poller service.

Reply:
If MonitoringEndTime.UTC() defaults to 0001-01-01 00:00:00 +0000 UTC, then I would assume there is some issue with Kruize, in that case I don't think we should commit the msg.
However I am happy to add a warning log before `return`, does that sound good?

Reply:
I will add a warn log as well in the else, thanks!

Reply:
Change pushed